src/components/organisms/UiMenu/_internal/UiMenuItem.vue

```diff
@@ -1,97 +1,89 @@
-import type { ListItemAttrsProps } from '../../UiList/_internal/UiListItem.vue';
-import UiMenuItemSuffix from './UiMenuItemSuffix.vue';
-import { useAttributes } from '../../../../composable';
 import type { DefineAttrsProps } from '../../../../types';
 import type { MenuItemProps } from './MenuItemProps';
-import type { MenuItem } from '../UiMenu.vue';
 
-export type MenuItemAttrsProps = DefineAttrsProps<MenuItemProps, ListItemAttrsProps>;
-export type ListItemInstance = InstanceType<typeof UiListItem>
+export type MenuItemAttrsProps = DefineAttrsProps<MenuItemProps>;
+
+defineOptions({ inheritAttrs: false });
 
-const props = withDefaults(defineProps<MenuItemProps>(), {
-  icon: 'present',
-  suffixVisible: 'default',
-  suffixAttrs: () => ({ class: 'ui-button--text ui-menu-item__suffix' }),
-  listItemAttrs: () => ({ class: 'ui-menu-item' }),
+const attrs:ListItemAttrsProps = useAttrs();
+const attrsForListItem = computed(() => {
+  const {
+    name, label, ...rest
+  } = attrs;
+  return rest;
 });
-const { attrs } = useAttributes();
-const isSelected = computed(() => (attrs.value.class && attrs.value.class.includes('ui-menu-item--is-selected')));
-const hasSuffix = computed(() => !!(props.suffixVisible === 'always' || (props.suffixVisible === 'default' && isSelected.value)));
-const buttonClass = computed(() => ({ 'ui-button--is-selected': isSelected.value }));
+
+const isSelected = computed(() => (('class' in attrsForListItem.value)
+  ? attrs.class.includes('ui-menu-item--is-selected')
+  : false));
+const hasSuffix = computed(() => (isSelected.value
+    || attrsForListItem.value.hasSuffix));
+const isFocused = ref(false);
+const tabindex = ref(0);
+const handleMenuItemFocus = () => { isFocused.value = true; };
+const handleMenuItemBlur = () => { isFocused.value = false; };
 const defaultProps = computed(() => ({
-  suffixAttrs: {
-    icon: props.icon,
-    class: 'ui-button--text ui-menu-item__suffix',
-    ...props.suffixAttrs,
-  },
   listItemAttrs: {
-    class: 'ui-menu-item',
-    ...props.listItemAttrs,
+    class: [
+      'ui-menu-item',
+      { 'ui-menu-item--is-selected': isSelected.value },
+    ],
+    ...{ ...(attrs.listItemAttrs || {}) },
+  },
+  suffixAttrs: {
+    icon: attrsForListItem.value.icon || 'present',
+    ...attrsForListItem.value.suffixAttrs,
   },
 }));
-const tabindex = ref(null);
-defineExpose({ tabindex });
-const menuItem = ref<ListItemInstance | null>(null);
-const menuItems = inject<Ref<MenuItem[]>>('menuItems', ref([]));
-onMounted(async () => {
-  await nextTick();
-  if (!menuItem.value) return;
-  menuItems.value = [
-    ...menuItems.value,
-    {
-      $el: menuItem.value.$el,
-      tabindex,
-    },
-  ];
+const menuItemClass = computed(() => ([
+  'ui-menu-item__content',
+  'ui-button--outlined',
+  { 'ui-button--is-selected': isSelected.value },
+]));
+
+const menuItemTemplateRefs = ref<InstanceType<typeof UiListItem> | null>(null);
```

Comment:
And here again. If it's a `...Refs` (plural), then why is this an `InstanceType<typeof UiListItem> | null>` and not an array?